src/NHibernate/Dialect/Dialect.cs

```diff
@@ -799,7 +818,67 @@ public virtual string GetIfExistsDropConstraint(Table table, string name)
 		/// <param name="table">The table.</param>
 		/// <param name="name">The name.</param>
 		/// <returns></returns>
+		// Since v5.1
+		[Obsolete("Can cause issues when a custom schema is defined (https://nhibernate.jira.com/browse/NH-1285). The new overload with the defaultSchema parameter should be used instead")]
 		public virtual string GetIfExistsDropConstraintEnd(Table table, string name)
+		{
+			var catalog = table.GetQuotedCatalog(this, null);
+			var schema = table.GetQuotedSchema(this, null);
+			var tableName = table.GetQuotedName(this);
+
+			return GetIfExistsDropConstraintEnd(catalog, schema, tableName, name);
+		}
+
+		/// <summary>
+		/// The syntax that is used to check if a constraint does not exists before creating it
+		/// </summary>
+		/// <param name="catalog">The catalog.</param>
+		/// <param name="schema">The schema.</param>
+		/// <param name="table">The table.</param>
+		/// <param name="name">The name.</param>
+		/// <returns></returns>
+		public virtual string GetIfNotExistsCreateConstraint(string catalog, string schema, string table, string name)
+		{
+			return "";
```

Comment:
I realize a bit late that we have a breaking change here.
`virtual string GetIfNotExistsCreateConstraint(Table table, string name)` has been obsoleted in favor of this new `virtual string GetIfNotExistsCreateConstraint(string catalog, string schema, string table, string name)` method, which is now used instead of the old in NHibernate code base.

So dialects overriding the obsolete method will cease working as expected, because their override will now just be ignored. It means that such dialects will return an empty string for the new `GetIfNotExistsCreateConstraint`, since they do not override it. And so such dialects will no more be able to create constraints.

So when obsoleting overridable methods, we cause a breaking change by ceasing using them.

In the present case, since the new default implementation is indeed "no-op", we could try avoiding the breaking change by calling the old obsolete method instead. The old method should then, of course, be reverted to its old "no-op" behavior instead of calling the new implementation. So furthermore all its removed overrides would have to be put in place again for avoiding another breaking change.

But this does not stops here. If we do that, how should we then consider overrides of the new method? These overrides would then be somewhat breaking change too, since they would cause another dialect further down the inheritance chain and overriding itself the obsolete method to cease achieving its override...
But in the case of this PR, removing these new overrides cannot be done without undoing the PR.

So either we fully revert this PR to avoid this kind of breaking change, or we add this as possible breaking change in 5.1, or we bump version to 6.0 (and still add this as possible breaking change).

An additional conclusion to this is that obsoleting overridable methods is in most cases a breaking change. (It could be done without breaking changes only if the new method call the old one and is not overridden in an exposed descendant, unless the new override also call the old method.)